DXRModules/DeusEx/Classes/DXRFlagsNGPMaxRando.uc

```diff
@@ -181,38 +183,45 @@ function NewGamePlus()
     p.saveCount=0;
     exp = 1;
     randomStart = settings.starting_map;
-    bingo_win = settings.bingo_win;
-    bingo_freespaces = settings.bingo_freespaces;
 
     // always enable maxrando when doing NG+?
     maxrando = 1;
     if(maxrando > 0) {
         // rollback settings to the default for the current difficulty
         // we only want to do this on maxrando because we want to retain the user's custom choices
-        SetDifficulty(difficulty);
+        bingo_win = settings.bingo_win;
+        bingo_freespaces = settings.bingo_freespaces;
+        newgameplus_curve_scalar = moresettings.newgameplus_curve_scalar;
+        CombatDifficulty = p.CombatDifficulty;
         ExecMaxRando();
+        settings.bingo_win = bingo_win;
+        settings.bingo_freespaces = bingo_freespaces;
+        moresettings.newgameplus_curve_scalar = newgameplus_curve_scalar;
+        SetDifficulty(difficulty);
+
         // increase difficulty on each flag like exp = newgameplus_loops; x *= 1.2 ^ exp;
         exp = newgameplus_loops;
     }
 
+    oldseed = dxr.SetSeed(seed - newgameplus_loops);
+    NewGamePlusValFloat(p.CombatDifficulty, 1.3, exp, 0, 15); // Anything over 15 is kind of unreasonably impossible
+    NewGamePlusValInt(settings.minskill, 1.1, exp, 10, 400);
+    NewGamePlusValInt(settings.maxskill, 1.1, exp, 10, 700);
+    NewGamePlusValInt(settings.enemiesrandomized, 1.2, exp, 10, 1000);
+    NewGamePlusValInt(settings.enemystats, 1.2, exp, 5, 100);
+    NewGamePlusValInt(settings.hiddenenemiesrandomized, 1.2, exp, 10, 1000);
+    NewGamePlusValInt(settings.ammo, 0.9, exp, 5, 100);
+    NewGamePlusValInt(settings.medkits, 0.9, exp, 5, 100);
+    NewGamePlusValInt(settings.multitools, 0.9, exp, 5, 100);
+    NewGamePlusValInt(settings.lockpicks, 0.9, exp, 5, 100);
+    NewGamePlusValInt(settings.biocells, 0.9, exp, 5, 100);
+    NewGamePlusValInt(settings.medbots, 0.9, exp, 3, 100);
+    NewGamePlusValInt(settings.repairbots, 0.9, exp, 3, 100);
+    NewGamePlusValInt(settings.turrets_add, 1.3, exp, 3, 1000);
+    NewGamePlusValInt(settings.merchants, 0.9, exp, 5, 100);
+    dxr.SetSeed(oldseed);
```

Comment:
You don't need this call to SetSeed since we have SetGlobalSeed("NewGamePlus"); right below it, which means we don't need the oldseed variable at all

Reply:
That's what I suspected but I was trying to be as safe as possible.